### ISSUE TITLE
Users added to "Teams" are automatically added to watchers on all repos in organization

### ISSUE THREAD

**Issue by sgabenov** [go-gitea/gitea#23892]
### Description

When you add new user to Team, this user is automatically subscribed to watch on all repos in the organization. 
The planed behavior is to add user WITHOUT watching all repos.

Also, it would be great, if there was a function to manage subscriptions by Admin for managed users, so you can in batch operation subscribe or unsubsribe users to watch repos.

### Gitea Version

1.19.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### How are you running Gitea?

docker

### Database

PostgreSQL

**Comment by silverwind** [comment]
Do we know how this works on other Forges like GitHub or GitLab? I'm kind of torn on this one as sometimes these auto-watches are useful, but I do recall complaints about them too.

**Comment by BElluu** [comment]
I'm pretty sure that's not how it works on github. However, for some it may be a feature and for others a bug, so I would parameterize it in the organization's settings.

**Comment by silverwind** [comment]
Yes, per-org configuration would be ideal, and a default in `app.ini` that applies to new orgs.

**Comment by melmus** [comment]
I agree with author this issue.
Can you add some option in app.ini to fix problem?

**Comment by gabriel-vasile** [comment]
There is a app.ini setting for this but it seems to have 2 purposes:

1. tell if new repo should be automatically watched by members
2. tell if new member should automatically watch all repos

https://github.com/go-gitea/gitea/blob/92e07f270aa925d518b13686f78befb63da0a747/custom/conf/app.example.ini#L874-L877
https://github.com/go-gitea/gitea/blob/92e07f270aa925d518b13686f78befb63da0a747/models/org_team.go#L504-L518

**Comment by ddi-acassidy** [comment]
per-user would be nice as well. Not all users need or want notifications for everything in the organization

**Comment by AndersBennedsgaard** [comment]
I fully agree with @ddi-acassidy . It would be nice with something similar to the notifications tab in Github: https://github.com/settings/notifications

**Comment by lunny** [comment]
So I think this issue proposal a per-user level notification setting UI but not a bug.

**Comment by AndersBennedsgaard** [comment]
I forgot to mention that I created https://github.com/go-gitea/gitea/issues/30537, which is a proposal for per-user notification settings.

I don't think this is a duplicate, due @sgabenov's proposal for 

> [...] a function to manage subscriptions by Admin for managed users, so you can in batch operation subscribe or unsubsribe users to watch repos.

unless we think that https://github.com/go-gitea/gitea/issues/30537 solves that

**Comment by zc-devs** [comment]
It goes funny when you set users as `hidden`:

![Screenshot 2024-06-23 134742](https://github.com/go-gitea/gitea/assets/127358482/7888b5b0-0fd4-40e3-a471-465bf9ac1be1)
![Screenshot 2024-06-23 134852](https://github.com/go-gitea/gitea/assets/127358482/1c0b1724-ed3d-42f3-afd9-41be0c9f28f1)

Create test repository:
![Screenshot 2024-06-23 135028](https://github.com/go-gitea/gitea/assets/127358482/3db7ab36-ae32-47e0-90ff-549b8f7deb7c)

Gitea `1.22.0`, `1.21.11`

**Comment by lunny** [comment]
Watchers are not team members or organization members.
